components/Ethdriver/src/ethdriver.c

```diff
@@ -298,6 +298,22 @@ int client_rx(int *len)
     return ret;
 }
 
+int client_has_rx(void)
+{
+    if (!done_init) {
+        return -1;
+    }
+
+    client_t *client = eth_get_client();
+    assert(client);
+
+    if ((client->pending_rx_head != client->pending_rx_tail) &&
+        (0 == client->should_notify)) {
```

Comment:
Why must `client->should_notify` be clear here?

Reply:
Oh man, I wasn't the one who originally wrote this code, I'm just responsible for creating the PR :)

My thought is that the `should_notify` flag is cleared after the Ethdriver signals the client. So if the flag is cleared, the client has already been signalled and the client is free to check for data.

Reply:
I wonder if this a performance optimization that assume the client has a certain state machine? What would happen in your client if you do not check this flag, but always get the "raw" state of the rx data. Seems it would read data then here already instead of doing so in the loop that waits for the notification?

I think it would be good to have some explanation in the code why things are done that are not trivial to understand.

Reply:
Honestly, I'm wondering if this was implemented in a system where the client couldn't tell who was contacting it. The client now has a specific handler for the Ethdriver, so this would just be an extra check.

I'll leave this up for community discussion to see if this is an interface that others are interested in.

Reply:
Currently, it's not possible to check if a packet is available using the rpc interface other than by calling `rx` which would also consume the packet if it's available and return a code indicating no packet is available otherwise. If packets are available the return code of rx indicates whether the next packet is available. A signal is sent to the client when the first packet becomes available. 

Internally, `client->should_notify` is set after the client is told there are no more packets available.  It is then cleared after a notification is sent.

The interface expects that the client can use the notification to know if packets are available and so this client_has_rx function isn't necessary. 

@chrisguikema, if you don't need this functionality any more, my vote would be on dropping the check from your code as it would unnecessarily increase the packet processing latency.

Reply:
Will do. Thanks for the feedback. This was also part of our backlog, so the feedback that it isn't necessary is appreciated, especially since it was originally written so long ago.